File: client/src/app/entities/print.ts
import {User} from './user';
import {Printorder} from './printorder';
import {Printstatus} from './printstatus';
import {DataPage} from '../shared/data-page';

export class Print {
  id: number;
  code: string;
  printorder: Printorder;
  sdate: string;
  edate: string;
  printstatus: Printstatus;
  creator: User;
  tocreation: string;
}

export class PrintDataPage extends DataPage{
    content: Print[];
}

File: server/src/main/java/bit/project/server/entity/Print.java
package bit.project.server.entity;

import lombok.Data;
import javax.persistence.*;
import java.time.LocalDate;
import javax.persistence.Id;
import java.time.LocalDateTime;
import lombok.NoArgsConstructor;
import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import com.fasterxml.jackson.annotation.JsonInclude;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

@Data
@Entity
@NoArgsConstructor
@JsonInclude(JsonInclude.Include.NON_NULL)
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class Print{
    @Id
    @GeneratedValue(strategy=GenerationType.IDENTITY)
    private Integer id;

    private String code;

    private LocalDate sdate;

    private LocalDate edate;

    private LocalDateTime tocreation;


    @ManyToOne
    private Printorder printorder;

    @ManyToOne
    private Printstatus printstatus;

    @ManyToOne
    @JsonIgnoreProperties({"creator","status","tocreation","roleList"})
    private User creator;


    public Print(Integer id){
        this.id = id;
    }

    public Print(Integer id, String code, Printorder printorder){
        this.id = id;
        this.code = code;
        this.printorder = printorder;
    }

}